### ISSUE TITLE
Cannot Parse Proto from JSON

### ISSUE THREAD

**Issue by BenRKarl** [googleapis/proto-plus-python#37]
The canonical (I think) way of parsing a proto from JSON as documented [here](https://googleapis.dev/python/protobuf/latest/google/protobuf/json_format.html#google.protobuf.json_format.Parse) doesn't work with the wrapped protos from this library. Here's a repro case:

```py
import proto
from google.protobuf.json_format import MessageToJson, Parse

class Foo(proto.Message):
    bar = proto.Field(proto.INT64, number=1)

foo = Foo(bar=22)

json = MessageToJson(foo._pb)

foo2 = Parse(json, Foo())
# pb_type = self._meta.fields[key].pb_type
# KeyError: 'DESCRIPTOR'
```

Strangely using the recommended pattern from this separate issue (https://github.com/googleapis/proto-plus-python/issues/32) yields unexpected results:

```py
foo2 = Parse(json, type(foo).pb())
# <class 'Foo'> not a deserialized instance, but the constructor
foo2 = Parse(json, Foo.pb())
# <class 'Foo'> same as above
```

**Comment by software-dov** [comment]
Parse is not expecting a proto-plus wrapped type. The following works without issue, although it isn't what you're trying to do:
```python
foo = Foo(bar=22)
json = MessageToJson(foo._pb)
foo2 = Foo(Parse(json, Foo()._pb))
```

**Comment by software-dov** [comment]
There are a couple potential solutions, ranked in no particular order of preference:
1. Add a class-only method to parse from json, either using the above hack or doing something more correct/performant.
2. Make a PR to google.protobuf to change `json_format.Parse` to check for proto-plus.Message
3. Do the pattern by hand.

I am heavily in favor of 1, as well as adding converting to json methods. The point of this repo is to abstract and hide google.protobuf as much as possible, and 3 is user hostile.

**Comment by BenRKarl** [comment]
I'm definitely in favor of a correct/performant class-only method. It doesn't make sense that json_format.Parse should be type checking and having something non-manual would be nice for everyone.

Perhaps the interface change is similar enough to this issue (https://github.com/googleapis/proto-plus-python/issues/32) that they can be bundled together? I think part of the awkwardness comes from the way in which the raw messages are exposed.

**Comment by BenRKarl** [comment]
Also is there a reason the method would be class-only and not accessible on an instance? The Ads library has a mechanism that abstracts message class instantiation so accessing class methods means adding additional module imports. And importing our message class isn't a) well documented and b) not recommended because the paths can change by virtue of the code being generated.

**Comment by software-dov** [comment]
User provided name collisions. We have to be extremely conservative about what fields and methods we add to instances so that there is no possibility of a user generating a field with that same name. Lifting functionality into class-only methods punts that responsibility.

**Comment by BenRKarl** [comment]
Ahhh yes field collisions. 

I suppose it's possible to get around an additional import by using `type(message_instance).method`

In any case we can determine the best method and document it clearly.

**Comment by software-dov** [comment]
Implemented via #39
